### DISCUSSION TITLE
"ruby -W1 --disable=did_you_mean,rubyopt /usr/local/Homebrew/Library/Homebrew/brew.rb cleanup"

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#1032] [Getting started]
`brew upgrade` `brew doctor` `brew cleanup` causes `ruby -W1 --disable=did_you_mean,rubyopt /usr/local/Homebrew/Library/Homebrew/brew.rb cleanup` to appear in macOS Terminal Title Bar. 

Is this a problem? Do I need to correct an underlying issue?

Terminal Profile is set to show:
`Working directory or document`
`Active process name, arguments`
`TTY name`
```
$ brew config
HOMEBREW_VERSION: 3.0.5-18-g433cd83
ORIGIN: https://github.com/Homebrew/brew
HEAD: 433cd83cbc186afbe970289557ceee3f33c7383a
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7e1d1173183895f87c9f6742548263fe373ee980
Core tap last commit: 7 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine", "--appdir=/Applications", "--fontdir=/Library/Fonts", "--qlplugindir=/Library/QuickLook", "--internet_plugindir=/Library/Internet Plug-Ins", "--prefpanedir=/L>", "--servicedir=/Library/Services"]
HOMEBREW_EDITOR: nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.30.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.32.29
Xcode: 12.4
```

https://user-images.githubusercontent.com/9438406/110848472-c1405200-8273-11eb-8114-88d2a284da06.mov

**Comment by gromgit** [comment]
Not an issue. Most `brew XYZ` subcommands resolve to `ruby -W1 --disable=did_you_mean,rubyopt /usr/local/Homebrew/Library/Homebrew/brew.rb XYZ`, and since you set your terminal to display `Active process name, arguments`, it's not surprising that appears in your titlebar.

**Reply by ghost** [reply]
Thank you for the explanation. The phrase "did you mean" makes it sound like the brew command was directed to an incorrect path.

**Reply by gromgit** [reply]
Ah, that would be the standard-issue [`did_you_mean` Ruby gem](https://github.com/ruby/did_you_mean), which I guess would be useful during development, but would just confuse Homebrew end-users with irrelevant suggestions.

**Reply by ghost** [reply]
Adrian, thank you for the background info. One last question on the difference in behavior here....

In Terminal when I use `login admin`  from the user account "John"  and run the same brew commands, I never see `ruby -W1 --disable=did_you_mean,rubyopt /usr/local/Homebrew/Library/Homebrew/brew.rb` in the Terminal title bar (Terminal Prefs are exactly the same for both users).

I see `bsdtar ◂ ruby` in the Terminal Title bar.

The shell still has my current logged in username, "John", because I am running it from account John, but I have used `login` to run the brew commands as another user "Admin" in that shell.

1. I installed homebrew using my Admin account but usually `update` and `upgrade` using my other user account using `login admin` in Terminal.

**Reply by gromgit** [reply]
I don't know how Terminal figures out when to update its own titlebar with process info, but it's likely that commands like `brew update` and `brew upgrade` quickly get to the point when they start spawning external commands like `bsdtar` to Get Things Done.

In contrast, commands like `brew cleanup` take longer to figure out what to do and/or do most of their work in Ruby itself, so you get to look at the underlying command for longer.
